### PR TITLE
Add external predicate to matchEquals and matchAny in ReceiverBu…

### PR DESCRIPTION
**Pull request by He-Pin**
refs https://github.com/akka/akka/issues/25774

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowPrependSpec.scala [hunk 1]
```diff
@@ -5,8 +5,6 @@
 package akka.stream.scaladsl
 
 //#prepend
-import akka.stream.scaladsl.Source
-import akka.stream.scaladsl.Sink
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowPrependSpec.scala]: I guess there is a warning for this, but they are here to show the imports in the docs.
Please revert.

**Review by patriknw — COMMENTED** [review]
looking good apart from change in unrelated file

**Comment by hepin1989** [comment]
```
[11-05 18:29:36.930] [info] akka.io.dns.AsyncDnsResolverIntegrationSpec *** ABORTED *** (10 seconds, 284 milliseconds)
[11-05 18:29:36.930] [info]   The code passed to eventually never returned normally. Attempted 51 times over 5.065460801 seconds. Last failure message: "" did not include substring "all zones loaded". (DockerBindDnsService.scala:60)
[11-05 18:29:36.973] [info]   org.scalatest.exceptions.TestFailedDueToTimeoutException:
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.tryTryAgain$1(Eventually.scala:432)
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.eventually(Eventually.scala:439)
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.eventually$(Eventually.scala:391)
[11-05 18:29:36.974] [info]   at akka.io.dns.AsyncDnsResolverIntegrationSpec.eventually(AsyncDnsResolverIntegrationSpec.scala:24)
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.eventually(Eventually.scala:337)
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.eventually$(Eventually.scala:336)
[11-05 18:29:36.974] [info]   at akka.io.dns.AsyncDnsResolverIntegrationSpec.eventually(AsyncDnsResolverIntegrationSpec.scala:24)
[11-05 18:29:36.974] [info]   at akka.io.dns.DockerBindDnsService.atStartup(DockerBindDnsService.scala:60)
[11-05 18:29:36.974] [info]   at akka.io.dns.DockerBindDnsService.atStartup$(DockerBindDnsService.scala:27)
[11-05 18:29:36.974] [info]   at akka.io.dns.AsyncDnsResolverIntegrationSpec.atStartup(AsyncDnsResolverIntegrationSpec.scala:24)
[11-05 18:29:36.974] [info]   at akka.testkit.AkkaSpec.beforeAll(AkkaSpec.scala:82)
[11-05 18:29:36.974] [info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:212)
[11-05 18:29:36.974] [info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
[11-05 18:29:36.974] [info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[11-05 18:29:36.974] [info]   at akka.testkit.AkkaSpec.run(AkkaSpec.scala:60)
[11-05 18:29:36.974] [info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[11-05 18:29:36.974] [info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:507)
[11-05 18:29:36.974] [info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:304)
[11-05 18:29:36.974] [info]   at sbt.ForkMain$Run$$Lambda$280/261845810.call(Unknown Source)
[11-05 18:29:36.974] [info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[11-05 18:29:36.974] [info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[11-05 18:29:36.974] [info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[11-05 18:29:36.974] [info]   at java.lang.Thread.run(Thread.java:745)
[11-05 18:29:36.974] [info]   Cause: org.scalatest.exceptions.TestFailedException: "" did not include substring "all zones loaded"
[11-05 18:29:36.974] [info]   at org.scalatest.MatchersHelper$.indicateFailure(MatchersHelper.scala:346)
[11-05 18:29:36.974] [info]   at org.scalatest.Matchers$ShouldMethodHelper$.shouldMatcher(Matchers.scala:6668)
[11-05 18:29:36.974] [info]   at org.scalatest.Matchers$AnyShouldWrapper.should(Matchers.scala:6704)
[11-05 18:29:36.974] [info]   at akka.io.dns.DockerBindDnsService.$anonfun$atStartup$1(DockerBindDnsService.scala:61)
[11-05 18:29:36.974] [info]   at akka.io.dns.DockerBindDnsService$$Lambda$4523/758788453.apply(Unknown Source)
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.makeAValiantAttempt$1(Eventually.scala:395)
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.tryTryAgain$1(Eventually.scala:409)
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.eventually(Eventually.scala:439)
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.eventually$(Eventually.scala:391)
[11-05 18:29:36.974] [info]   at akka.io.dns.AsyncDnsResolverIntegrationSpec.eventually(AsyncDnsResolverIntegrationSpec.scala:24)
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.eventually(Eventually.scala:337)
[11-05 18:29:36.974] [info]   at org.scalatest.concurrent.Eventually.eventually$(Eventually.scala:336)
[11-05 18:29:36.974] [info]   at akka.io.dns.AsyncDnsResolverIntegrationSpec.eventually(AsyncDnsResolverIntegrationSpec.scala:24)
[11-05 18:29:36.974] [info]   at akka.io.dns.DockerBindDnsService.atStartup(DockerBindDnsService.scala:60)
[11-05 18:29:36.974] [info]   at akka.io.dns.DockerBindDnsService.atStartup$(DockerBindDnsService.scala:27)
[11-05 18:29:36.974] [info]   at akka.io.dns.AsyncDnsResolverIntegrationSpec.atStartup(AsyncDnsResolverIntegrationSpec.scala:24)
[11-05 18:29:36.975] [info]   at akka.testkit.AkkaSpec.beforeAll(AkkaSpec.scala:82)
[11-05 18:29:36.975] [info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:212)
[11-05 18:29:36.975] [info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
[11-05 18:29:36.975] [info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[11-05 18:29:36.975] [info]   at akka.testkit.AkkaSpec.run(AkkaSpec.scala:60)
[11-05 18:29:36.975] [info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[11-05 18:29:36.975] [info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:507)
[11-05 18:29:36.975] [info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:304)
[11-05 18:29:36.975] [info]   at sbt.ForkMain$Run$$Lambda$280/261845810.call(Unknown Source)
[11-05 18:29:36.975] [info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[11-05 18:29:36.975] [info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[11-05 18:29:36.975] [info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[11-05 18:29:36.975] [info]   at java.lang.Thread.run(Thread.java:745)
```

**Comment by raboof** [comment]
That failure is indeed unrelated to this PR and tracked as #25855

**Comment by raboof** [comment]
PLS BUILD

**Comment by hepin1989** [comment]
@raboof Do you have any time to take another round of review?

File: akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java [hunk 2]
```diff
@@ -270,13 +294,28 @@ public boolean defined(Object o) {
...
-        return true;
-      }
-    }, apply));
+    addStatement(new UnitCaseStatement<>(ALWAYS_TRUE,
+                                         apply));
+    return this;
+  }
+
+  /**
+   * Add a new case statement to this builder, that matches any argument.
```
Comment (raboof) [inline akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java]: Any argument that matches the predicate, at least?

**Review by raboof — APPROVED** [review]
Nice

File: akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java (hunk 2 above)
Comment (hepin1989) [reply akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java]: True，will adjust it.

Comment (hepin1989) [reply akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java]: @raboof I have updated it, thanks.

**Comment by hepin1989** [comment]
refs: failed：akka.discovery.dns.DnsDiscoverySpec.Dns discovery with the system resolver must work with SRV records

**Comment by hepin1989** [comment]
refs: https://github.com/akka/akka/issues/26068

**Comment by raboof** [comment]
PLS BUILD

**Comment by hepin1989** [comment]
@patriknw Do you have any time :)

**Comment by helena** [comment]
Restarted all checks to see where this is.

**Comment by hepin1989** [comment]
Do I need to update something more?

**Review by helena — APPROVED** [review]
LGTM thanks :)

**Comment by patriknw** [comment]
My bad that I didn't communicate this earlier. Since we are in release candidate phase we shouldn't merge code changes to master unless it's fixing critical issues, followed by new RC.
Reverted in https://github.com/akka/akka/pull/28027

I will create a new PR in @hepin1989 name which we can merge after 2.6.0, but I'd also like to discuss if we really want to add more and more overloads the the builders. Let's do that there.

**Comment by patriknw** [comment]
New PR in https://github.com/akka/akka/pull/28029

**Pull request merged**